lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1810,18 +1805,73 @@ GLStatus TFGraphLowering::visitGraphOperationInst(GraphOperationInst *inst) {
         break;
       }
       case SymbolicValue::Array: {
-        if (isShapeArrayPseudoAttr(name, attrValue)) {
-          // SHAPE_ARRAY_ATTR is a pseudo-attribute used by the compiler's
-          // partitioning and graph lowering passes to propagate shape info for
-          // XLA compilation (e.g. feed shape info to infeed / outfeed ops), and
-          // will not be lowered into this graph op itself.
+        // SHAPE_ARRAY_ATTR is a pseudo-attribute used by the compiler's
+        // partitioning and graph lowering passes to propagate shape info for
+        // XLA compilation (e.g. feed shape info to infeed / outfeed ops), and
+        // will not be lowered into this graph op itself.
+        if (isShapeArrayPseudoAttr(name, attrValue))
+          break;
+
+        CanType elementType;
+        auto rawElements = attrValue.getArrayValue(elementType);
+        SmallVector<SymbolicValue, 4> elements;
+        elements.reserve(rawElements.size());
+        for (auto elt : rawElements)
+          elements.push_back(elt.lookThroughSingleElementAggregates());
+
+        auto elementTypeString = elementType->getString();
+
+        // TODO: TF_SetAttrTypeList
+
+        if (elementTypeString == "String") {
+          SmallVector<const void*, 4> pointers;
+          SmallVector<size_t, 4> sizes;
+          pointers.reserve(elements.size());
+          sizes.reserve(elements.size());
+          for (auto elt : elements) {
+            auto bytes = elt.getStringValue();
+            pointers.push_back(bytes.data());
+            sizes.push_back(bytes.size());
+          }
+          TF_SetAttrStringList(op, name.c_str(), pointers.data(), sizes.data(),
+                               elements.size());
           break;
-        } else {
-          // FIXME: Handle array attributes.
-          internalError(getUserSourceLocation(inst->getDebugLocation()),
-                        "FIXME: Handle array attributes");
-          return GLStatus::Error;
         }
+        if (StringRef(elementTypeString).startswith("Int")) {
+          SmallVector<int64_t, 4> values;
+          values.reserve(elements.size());
+          for (auto elt : elements)
+            values.push_back(elt.getIntegerValue().getLimitedValue());
+          TF_SetAttrIntList(op, name.c_str(), values.data(), values.size());
+          break;
+        }
+        if (elementTypeString == "Float" || elementTypeString == "Double") {
+          SmallVector<float, 4> values;
+          values.reserve(elements.size());
+          for (auto elt : elements) {
+            auto value = elt.getFloatValue();
+            // TensorFlow only supports float32 attributes, and isn't designed
+            // for high precision, so we force truncate down.
+            bool losesInfo = false;
```

Comment:
nit: Some of this could be refactored into helper functions. I guess we can do it later.